api/src/test/java/org/pmiops/workbench/auth/DelegatedUserCredentialsTest.java

```diff
@@ -0,0 +1,148 @@
+   */
+  private static PrivateKey privateKeyFromPkcs8(String privateKeyPem) throws IOException {
+    Reader reader = new StringReader(privateKeyPem);
+    PemReader.Section section = PemReader.readFirstSectionAndClose(reader, "PRIVATE KEY");
+    if (section == null) {
+      throw new IOException("Invalid PKCS8 data.");
+    }
+    byte[] bytes = section.getBase64DecodedBytes();
+    PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(bytes);
+    try {
+      KeyFactory keyFactory = SecurityUtils.getRsaKeyFactory();
+      PrivateKey privateKey = keyFactory.generatePrivate(keySpec);
+      return privateKey;
+    } catch (NoSuchAlgorithmException | InvalidKeySpecException exception) {
+      throw new IOException("Unexpected expcetion reading PKCS data", exception);
+    }
+  }
+
+  /**
+   * Given a JWT payload, create a self-signed JWT using an OAuth2 compliant algorithm. This method
+   * mocks out the behavior of the IAM Credentials API, which signs JWTs using Google-managed
+   * service account private keys.
+   */
+  private static String createSelfSignedJwt(JsonWebToken.Payload payload)
+      throws IOException, GeneralSecurityException {
+    JsonWebSignature.Header header = new JsonWebSignature.Header();
+    header.setAlgorithm("RS256");
+    header.setType("JWT");
+    header.setKeyId(SA_PRIVATE_KEY_ID);
+
+    return JsonWebSignature.signUsingRsaSha256(
+        privateKeyFromPkcs8(SA_PRIVATE_KEY_PKCS8),
+        JacksonFactory.getDefaultInstance(),
+        header,
+        payload);
+  }
+
+  @Test
+  public void testClaims() {
+    JsonWebToken.Payload payload = creds.createClaims();
+
+    assertThat(payload.getAudience()).isEqualTo(GoogleOAuthConstants.TOKEN_SERVER_URL);
+    assertThat(payload.getIssuer()).isEqualTo(SERVICE_ACCOUNT_EMAIL);
+    assertThat(payload.getSubject()).isEqualTo(USER_EMAIL);
+    assertThat(payload.get("scope")).isEqualTo(String.join(" ", SCOPES));
+  }
+
+  @Test
+  public void testRefreshFlow() throws IOException {
+    // Mock out the IAM Credentials API client to create a self-signed JsonWebSignature instead of
+    // calling Google's API.
+    when(mockIamCredentialsClient.signJwt(any(SignJwtRequest.class)))
+        .then(
+            invocation -> {
+              SignJwtRequest request = invocation.getArgument(0);
+              JsonWebToken.Payload payload =
+                  DelegatedUserCredentials.JSON_FACTORY.fromInputStream(
+                      new ByteArrayInputStream(request.getPayload().getBytes()),
+                      JsonWebToken.Payload.class);
+              return SignJwtResponse.newBuilder()
+                  .setSignedJwt(createSelfSignedJwt(payload))
+                  .build();
+            });
+    // Register the expected service account & access token with the mock token server transport.
+    mockTokenServerTransport.addServiceAccount(SERVICE_ACCOUNT_EMAIL, "access-token");
+
+    // Kick off the refresh flow.
+    creds.refresh();
+
```

Comment:
Is there anything else you can mock & verify? Looks like all we verify is that the token value is preserved.